core/indigo-core/common/base_cpp/os_sync_wrapper.h

```diff
@@ -19,59 +19,17 @@
 #ifndef __os_sync_wrapper_h__
 #define __os_sync_wrapper_h__
 
+#include <mutex>
+#include <condition_variable>
+
 #include "base_c/defs.h"
-#include "base_c/os_sync.h"
 #include "base_cpp/exception.h"
 
 namespace indigo
 {
 
-    // os_mutex wrapper
-    class DLLEXPORT OsLock
-    {
-    public:
-        OsLock();
-        ~OsLock();
-
-        void Lock();
-        void Unlock();
-
-    private:
-        os_mutex _mutex;
-    };
-
-    // Automatic lock/unlock
-
-    template <typename T, bool lock_can_be_null> class OsLockerT
-    {
-    public:
-        OsLockerT(T* lock) : _lock(lock)
-        {
-            if (_lock != NULL)
-                _lock->Lock();
-            else if (!lock_can_be_null)
-                throw Exception("Passed lock object pointer is NULL");
-        }
-
-        OsLockerT(T& lock) : _lock(&lock)
-        {
-            _lock->Lock();
-        }
-
-        ~OsLockerT()
-        {
-            if (_lock != NULL)
-                _lock->Unlock();
-        }
-
-    private:
-        T* _lock;
-    };
-    typedef OsLockerT<OsLock, false> OsLocker;
-    typedef OsLockerT<OsLock, true> OsLockerNullable;
```

Comment:
Replace OsLockerNullable with std::unique_lock. (64d47b47bcb335575f1c46964975c21832ef0f6d)
resulted in increasing code in 6 places. So, some macro/template instead this would be useful.